docs/Cours fondamentaux ML/7 - Tradeoff biais-variance.ipynb

```diff
@@ -0,0 +1,459 @@
+    "\n",
+    "Et non ! Un [adage très connu en statistique](https://fr.abcdef.wiki/wiki/All_models_are_wrong)\n",
+    "décrit la situation ainsi : **Tous les modèles sont faux, mais certains sont utiles.**\n",
+    "\n",
+    "Il est impossible en pratique de parfaitement modéliser la relation $(x, y)$. Et on ne parle pas que du bruit irréductible,\n",
+    "mais même de la fonction réelle qui décrit au mieux la relation entre $x$ et la valeur moyenne $\\hat y$ sachant $x$.\n",
+    "Il faudrait pouvoir toutes les données possibles (pouvez-vous me trouver toutes les photos de chats et de chiens possibles ?),\n",
+    "ou faire pile poil les bonnes hypothèses sur la relation afin d'injecter les bons biais au modèle.\n",
+    "\n",
+    "Enfin, il est intéressant de voir qu'une partie du sur-apprentissage est en fait dû au modèle qui s'affine sur le bruit. Il aura alors\n",
+    "réduit les erreurs liées au bruit du jeu d'entraînement, mais il aura appris n'importe quoi !"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "576a4fa1-aff0-419e-aa6e-5d684e3bdfb1",
+   "metadata": {},
+   "source": [
+    "### Visualisation\n",
+    "On peut imaginer ce que donne un tracé de l'erreur d'un modèle dont on ferait progressivement augmenter\n",
+    "la complexité.\n",
+    "Petit à petit, l'erreur liée à son biais diminue, mais celle liée à la variance augmente.\n",
+    "On peut distinguer alors les deux régimes de sous-apprentissage et de sur-apprentissage.\n",
+    "\n",
+    ":::{figure-md} biais-variance-fig\n",
+    "<img src=\"src/biais_variance.png\" alt=\"tradeoff biais-variance\">\n",
+    "\n",
+    "Visualisation du compromis biais-variance.\n",
+    "Inspiré de l'image libre de droit [ici](https://www.ncbi.nlm.nih.gov/books/NBK543534/figure/ch8.Fig3/).\n",
+    ":::\n",
+    "\n",
+    "En pratique, on préfère prendre un modèle faiblement biaisé et ajouter de la régularisation afin de réduire petit à petit le sur-apprentissage."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3cc12414-c66e-421f-9011-43c22c047911",
+   "metadata": {},
+   "source": [
+    "### (\\*\\*) Détails mathématiques\n",
+    "Soit :\n",
+    "* $x, y$ : Des couples de données où $x$ est un ensemble de features et $y$ la valeur à déterminer à partir du vecteur $x$.\n",
+    "Ces couples proviennent d'une distribution de probabilité $P$ quelconque.\n",
+    "* $D$ : Un jeu de données quelconque constitué de couples $(x_i, y_i)$.\n",
+    "$D$ est la réalisation d'un échantillonage de $P$. On note $y_i = y(x_i)$.\n",
+    "* $h$ : Un modèle de ML capable d'apprendre à partir d'un jeu de données $D$.\n",
+    "Si $h$ a été entraîné sur $D$, on le note $h_D$, et on note ses prédictions $h_D(x)$.\n",
+    "\n",
+    "On peut alors décomposer le loss moyen d'un modèle $h$ :\n",
+    "\n",
+    "$$\\begin{align}\n",
+    "\\text{loss}_{\\text{test}}(h) & = \\text{variance}(h) + \\text{biais}(h)^2 + \\text{bruit}\\\\\n",
```

Comment:
Comme dit précédemment, tu pourrais introduire l'équation dès l'introduction en expliquant brièvement l'intuition (réussir à minimiser l'erreur de généralisation du modèle revient à minimiser la somme de 3 erreurs ...). Tu peux te référer au bouquin de Géron à ce sujet [Hands-On Machine Learning with Scikit-Learn and TensorFlow Concepts, Tools, and Techniques to Build Intelligent Systems] dès p.126 .

<img width="774" alt="Capture d’écran 2022-04-23 à 16 36 34" src="https://user-images.githubusercontent.com/40428556/164910606-020c9fb5-c7f7-421e-8e97-e82d34f7c84b.png">